File: models/Employee.js
// models/Employee.js
module.exports = (sequelize, DataTypes) => {
    const Employee = sequelize.define('Employee', {
        user_id: {type: DataTypes.BIGINT, primaryKey: true},
        employee_id: {type: DataTypes.STRING(64), allowNull: false, unique: true},
        phone: {type: DataTypes.STRING(32), allowNull: false},
        address: {type: DataTypes.STRING(255), allowNull: false},
        fathers_name: {type: DataTypes.STRING(120), allowNull: false},
        aadhar_no: {type: DataTypes.STRING(32), allowNull: false, unique: true},
        date_of_birth: {type: DataTypes.DATEONLY, allowNull: false},
        work_position: {type: DataTypes.STRING(80), allowNull: false},
        created_at: {type: DataTypes.DATE, allowNull: false, defaultValue: sequelize.fn('SYSUTCDATETIME')},
        updated_at: {type: DataTypes.DATE, allowNull: false, defaultValue: sequelize.fn('SYSUTCDATETIME')},
    }, {
        tableName: 'employees',
        timestamps: false, // handled by triggers/defaults
        underscored: true
    });

    Employee.associate = (models) => {
        Employee.belongsTo(models.AuthUser, {foreignKey: 'user_id', as: 'User'});
    };

    return Employee;
};

File: models/employee.py
from sqlalchemy.orm import Mapped, mapped_column, relationship
from sqlalchemy import BigInteger, String, Date, DateTime, ForeignKey
from datetime import datetime
from .base import Base


class Employee(Base):
    __tablename__ = "employees"
    __table_args__ = {"schema": "dbo"}

    user_id: Mapped[int] = mapped_column(
        BigInteger,
        ForeignKey("dbo.users.user_id", ondelete="CASCADE"),
        primary_key=True,
    )
    employee_id: Mapped[str] = mapped_column(String(64), unique=True, nullable=False)
    phone: Mapped[str] = mapped_column(String(32), nullable=False)
    address: Mapped[str] = mapped_column(String(255), nullable=False)
    fathers_name: Mapped[str] = mapped_column(String(120), nullable=False)
    aadhar_no: Mapped[str] = mapped_column(String(32), unique=True, nullable=False)
    date_of_birth: Mapped[datetime] = mapped_column(Date, nullable=False)
    work_position: Mapped[str] = mapped_column(String(80), nullable=False)
    created_at: Mapped[datetime] = mapped_column(
        DateTime, default=datetime.utcnow, nullable=False
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime, default=datetime.utcnow, nullable=False
    )

    User = relationship("AuthUser", back_populates="Employee")
